nemo/collections/asr/parts/utils/asr_confidence_utils.py

```diff
@@ -0,0 +1,310 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from functools import partial
+from typing import List, Optional
+
+from omegaconf import DictConfig, OmegaConf
+
+from nemo.collections.asr.parts.utils.rnnt_utils import Hypothesis
+
+
+@dataclass
+class ConfidenceMethodConfig:
+    name: str = "entropy"
+    entropy_type: str = "tsallis"
+    temperature: float = 0.33
+    entropy_norm: str = "exp"
+
+    def __post_init__(self):
```

Comment:
Post Init does not work with Hydra, but oh well.

Reply:
Can this help us to allow post init in Hydra?
https://github.com/facebookresearch/hydra/issues/981#issuecomment-1303647547

Reply:
Let's not add Hydra hacks into the codebase. It's not very good long term. Use OmegaConf.to_object() if you must